VmAgent.Core/VmDirectories.cs

```diff
@@ -39,8 +39,8 @@ public VmDirectories(string rootPath)
             GsdkConfigRootFolderContainer = Path.Combine(TempStorageRootContainer, "Config");
             GsdkConfigFilePathContainer = Path.Combine(GsdkConfigRootFolderContainer, GsdkConfigFilename);
 
-            AgentStateFile = Path.Combine(TempStorageRootVm, "PlayFabVmAgentState");
-            AgentStateTempFile = Path.Combine(TempStorageRootVm, "PlayFabVmAgentState.tmp");
+            AgentStateFile = Path.Combine(PlayFabFolderOnPrimaryDrive, "PlayFabVmAgentState");
+            AgentStateTempFile = Path.Combine(PlayFabFolderOnPrimaryDrive, "PlayFabVmAgentState.tmp");
             MonitoringStateFile = Path.Combine(TempStorageRootVm, "PlayFabMonitoringState");
             MonitoringStateTempFile = Path.Combine(TempStorageRootVm, "PlayFabMonitoringState.tmp");
             HostConfigOverrideFile = Path.Combine(TempStorageRootVm, "HostConfigOverride.json");
```

Comment:
Great point. That's definitely something we can evaluate for a future PR. At this time, we can test with the VmAgent state file to prevent VmAgent restarting when the secondary drive is full.